modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -1051,6 +1051,14 @@ void ONNXImporter::populateNet(Net dstNet)
             }
             else
             {
+                // Scale layer allocate output with the first input shape
+                if (total(outShapes[node_proto.input(0)]) < total(outShapes[node_proto.input(1)]))
+                {
+                    std::string secondInp = node_proto.input(1);
+                    node_proto.set_input(1, node_proto.input(0));
+                    node_proto.set_input(0, secondInp);
```

Comment:
> node_proto.set_input()

Not sure that it is a good idea to hack original input structures.

It is better to create some own intermediate object (or create methods to fill operations instead of 1000 LOC function).